src/disco/shred/fd_shred_tile.c

```diff
@@ -0,0 +1,694 @@
+    fd_mux_publish( mux, sig, fd_laddr_to_chunk( ctx->shred_store_wksp, s34+3UL ), sizeof(fd_shred34_t), ctl, tsorig, tspub );
+
+  /* Compute all the destinations for all the new shreds */
+
+  fd_shred_t const * new_shreds[ FD_REEDSOL_DATA_SHREDS_MAX+FD_REEDSOL_PARITY_SHREDS_MAX ];
+  ulong k=0UL;
+  for( ulong i=0UL; i<set->data_shred_cnt; i++ )
+    if( !d_rcvd_test( set->data_shred_rcvd,   i ) )  new_shreds[ k++ ] = (fd_shred_t const *)set->data_shreds  [ i ];
+  for( ulong i=0UL; i<set->parity_shred_cnt; i++ )
+    if( !p_rcvd_test( set->parity_shred_rcvd, i ) )  new_shreds[ k++ ] = (fd_shred_t const *)set->parity_shreds[ i ];
+
+  ulong out_stride;
+  ulong max_dest_cnt[1];
+  fd_shred_dest_idx_t * dests;
+  if( FD_LIKELY( ctx->last_frag_was_shred ) ) {
+    out_stride = k;
+    dests = fd_shred_dest_compute_children( ctx->sdest, new_shreds, k, _dests, k, fanout, fanout, max_dest_cnt );
+  } else {
+    out_stride = 1UL;
+    *max_dest_cnt = 1UL;
+    dests = fd_shred_dest_compute_first   ( ctx->sdest, new_shreds, k, _dests );
+  }
+  FD_TEST( dests );
+
+  // FD_LOG_NOTICE(( "Computed destination for %lu shreds. max_dest_cnt=%lu", k, *max_dest_cnt ));
+
+  /* Send only the ones we didn't receive. */
+  for( ulong i=0UL; i<k; i++ ) for( ulong j=0UL; j<*max_dest_cnt; j++ ) send_shred( ctx, new_shreds[ i ], dests[ j*out_stride+i ] );
+
+}
+
+static inline void
+warmup_arp_cache( shred_tile_ctx_t * ctx ) {
+  FD_LOG_NOTICE(( "Trying to find route to 1.1.1.1 to warmup the arp cache" ));
+  fd_ip_t * ip = ctx->args->ip;
+  ulong tsorig = fd_frag_meta_ts_comp( fd_tickcount() );
+  while( 1 ) {
+    uchar mac[6];
+    uint  out_next_ip[1];
+    uint  out_ifindex[1];
+
+    fd_ip_route_fetch( ip );
+    fd_ip_arp_fetch  ( ip );
+    int res = fd_ip_route_ip_addr( mac, out_next_ip, out_ifindex, ip, 0x01010101 );
+    if( res==FD_IP_NO_ROUTE  ) FD_LOG_ERR(( "Routing is misconfigured" ));
+    if( res==FD_IP_SUCCESS   ) break;
+    if( res!=FD_IP_PROBE_RQD ) FD_LOG_ERR(( "Unicast address resolved to multicast/broadcast" ));
+
+    uchar * arp_packet = fd_chunk_to_laddr( ctx->net_wksp, ctx->net_chunk );
+    ulong arp_sz[1];
+    res = fd_ip_arp_gen_arp_probe( arp_packet, sizeof(fd_ip_arp_t), arp_sz, *out_next_ip, ctx->args->src_ip, ctx->args->src_mac );
+    if( res!=FD_IP_SUCCESS ) FD_LOG_ERR(( "Generation of arp probe failed" ));
+
+    ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+    ulong   ctl = fd_frag_meta_ctl( ctx->tile_idx, 1, 1, 0 );
+    ulong   sig = fd_disco_netmux_sig( 0U, 0U, SRC_TILE_SHRED, SRC_TILE_NET ); /* arp is not IP */
+    fd_mcache_publish( ctx->args->to_net_mcache, ctx->net_depth, ctx->net_seq, sig, ctx->net_chunk,
+        *arp_sz, ctl, tsorig, tspub );
+    ctx->net_seq   = fd_seq_inc( ctx->net_seq, 1UL );
+    ctx->net_chunk = fd_dcache_compact_next( ctx->net_chunk, *arp_sz, ctx->net_chunk0, ctx->net_wmark );
+
+    FD_SPIN_PAUSE(); /* TODO: Pause for longer than a split instant */
```

Comment:
Hmm. Maybe do this todo? SPIN_PAUSE until rdtsc hits expected tickcount

Reply:
This doesn't need to be that precise, so I used `fd_log_wallclock` instead, but yes